src/Features/CSharp/Portable/UpgradeProject/CSharpUpgradeProjectCodeFixProvider.cs

```diff
@@ -85,8 +85,11 @@ public override bool IsUpgrade(ParseOptions projectOptions, string newVersion)
             var parseOptions = (CSharpParseOptions)projectOptions;
             Contract.ThrowIfFalse(LanguageVersionFacts.TryParse(newVersion, out var parsedNewVersion));
 
+            var mappedVersion = parsedNewVersion.MapSpecifiedToEffectiveVersion();
+
             // treat equivalent versions (one generic and one specific) to be a valid upgrade
-            return parsedNewVersion.MapSpecifiedToEffectiveVersion() >= parseOptions.LanguageVersion &&
+            return mappedVersion >= parseOptions.LanguageVersion &&
+                mappedVersion < LanguageVersion.CSharp8 &&
```

Comment:
This isn't clear to me why this check si good.  What if the user is trying to use a C# 8.0 or above feature.  this check looks like it will prevent the user from getting the upgrade offer... can you clarify (or at least document in teh code what hte intent is here)?

Reply:
The code fix to _automatically_ upgrade a project is being removed from scenarios where the specification for the target runtime does not required that all of C# 8's features be supported. You will still be able to manually update the project if 1) you know you are providing a runtime that supports the features under a different support policy and/or 2) you are only using a subset of C# 8's language features which do not go outside the bounds of what the runtime supports. This is implemented for consistency with https://devblogs.microsoft.com/dotnet/an-update-to-c-versions-and-c-tooling/.

Reply:
Could we instead just give a warning about this?  Having to go manually do this is *super* annoying.  Heck, just even finding the option is a PITA.

Reply:
Requests for change to the design is outside the scope of this pull request. The pull request was implemented in accordance with a design decision spanning multiple teams, and cannot be altered outside of an agreement from the same set.

Reply:
Which set of teams decided on this?  I'm just trying to figure out the experience for the user here?  It seems like this would come across as one of those "the product feels broken" fixes.  Woudl it be worth going back to that group and suggesting that we could have a better approach here that balances needs better?  i.e. "instead of disabling, we can still offer, but warn.  this allows users to still get the feature when tehy want, and when it's going to be right for them, while allowing other users to know this may have requirements on them they have to consider first".

Reply:
> a project is being removed from scenarios where the specification for the target runtime does not required that all of C# 8's features be supported.

This is the part i'm not quite sure i understand.  The user-facing feature is "make my project work so that i can use this language feature".  it sounds like that's an issue now because we only update the lang version (but we leave the target runtime spec alone).  Would it not be more user friendly to also say "if you want to use this feature, you need this target-runtime, and we can move you to that if you want"?